include/benchmark/benchmark.h

```diff
@@ -1534,8 +1534,15 @@ class Fixture : public internal::Benchmark {
 #endif
 
 // Helper macro to create a main routine in a test that runs the benchmarks
+// Note the workaround for Hexagon simulator passing argc != 0, argv = NULL.
 #define BENCHMARK_MAIN()                                                \
   int main(int argc, char** argv) {                                     \
+    char arg0_default[] = "benchmark";                                  \
```

Comment:
can we wrap this in `BENCHMARK_OS_QURT`?  it's non-standard behaviour so i don't really want it leaking out everywhere.

Reply:
This is inside a macro definition, so `#ifdef` can't be used here. I can replicate the whole `#define BENCHMARK_MAIN()` definition inside `#ifdef` as a work-around, but it feels less elegant. What would you prefer?

Reply:
oh you're right, of course. blech. having the whole thing repeated is bad, and having this chunk in here that's only for QURT is bad.

i guess this is the least bad :/